main.nf

```diff
@@ -140,392 +165,666 @@ process fastqc {
  */
 
 process trim_galore {
-    tag "reads: $reads"
+
+    tag "$name"
 
     module 'bioinfo-tools'
     module 'FastQC'
     module 'cutadapt'
     module 'TrimGalore'
 
-    cpus 3
-    memory '3 GB'
-    time '8h'
+    cpus 2
+    memory { 3.GB * task.attempt }
+    time { 8.h * task.attempt }
+    errorStrategy { task.exitStatus == 143 ? 'retry' : 'terminate' }
+    maxRetries 3
+    maxErrors '-1'
 
-    publishDir "$results_path/trim_galore"
+    publishDir "${params.outdir}/trim_galore", mode: 'copy'
 
     input:
-    file(reads) from trimgalore_input
+    set val(name), file(reads:'*') from trimgalore_input
 
     output:
-    file '*fq.gz' into trimmed_reads, trimmed_reads_miRdeep2
-    file '*trimming_report.txt'
-    
+    file '*.gz' into trimmed_reads_bowtie, trimmed_reads_bowtie2
+    file '*trimming_report.txt' into trimgalore_results
+
     script:
-    /* Trim Galore should automatically detect the small RNA adapter sequence */
-    """
-    trim_galore --gzip --fastqc_args "-q" ${reads}
-    """
+    /* I'm using the latest version of trimgalore downloaded from github. The version on uppmax has a bug when handling smallRNA-seq data. */
+
+    single = reads instanceof Path
+
+    if(single) {
+      """
+      perl /home/chuanw/trimgalore/trim_galore.pl --small_rna --gzip $reads
+      """
+    } else {
+      """
+      perl /home/chuanw/trimgalore/trim_galore.pl --paired --small_rna --gzip $reads
```

Comment:
Yes I know - I don't mean that we shouldn't _sequence_ PE (though we shouldn't, when possible). I mean that we shouldn't _analyse_ PE. If we just discard read two then it becomes SE.